apps/calendar/test/marionette/lib/views/week.js

```diff
@@ -77,6 +83,153 @@ Week.prototype = {
     });
   },
 
+  /**
+   * This returns a boolean that is true if the hour square is fully visible.
+   *
+   * @param {Number} hour The hour to locate
+   * @returns {Boolean} Whether the square is fully visible.
+   */
+  _isHourDisplayed: function(hour) {
+    return this._cmpHourDisplayed(hour) === 0;
+  },
+
+  /**
+   * This method tries to find out where the hour square is located in the
+   * window.
+   *
+   * @param {Number} hour The hour to locate.
+   * @returns {Number} 0 if it's displayed, or the delta otherwise: negative
+   * if it's above or positive if it's below.
+   */
+  _cmpHourDisplayed: function(hour) {
+    var hourElt = this.hours[hour];
+    var mainScroll = this.scrollTop;
+    var hourOffset = hourElt.scriptWith(function(el) { return el.offsetTop; });
+    var mainClientHeight = this.main.rect().height;
+    var hourHeight = hourElt.rect().height;
+
+    if (hourOffset < mainScroll) {
+      return hourOffset - mainScroll;
+    }
+
+    var bottomHourOffset = hourOffset + hourHeight;
+    var bottomScroll = mainScroll + mainClientHeight;
+    if (bottomHourOffset > bottomScroll) {
+      return bottomHourOffset - bottomScroll;
+    }
+    return 0;
+  },
+
+  /**
+   * This method waits until we don't scroll.
+   *
+   * Note this doesn't seem to work properly when the view is just displayed.
+   * That's why waitForHourScrollEnd is still useful.
+   */
+  waitForNoScroll: function() {
+    var prevScrollTop = null;
+    this.client.waitFor(function() {
+      var currentScrollTop = this.scrollTop;
+      if (prevScrollTop !== null && prevScrollTop === currentScrollTop) {
+        return true;
+      }
+      prevScrollTop = currentScrollTop;
+      return false;
+    }.bind(this));
+  },
+
+  /**
+   * This method scrolls slowly to the right hour.
+   *
+   * @param {Number} hour The hour to scroll to.
+   */
+  scrollToHour: function(hour) {
+    this.waitForNoScroll();
+    if (this._isHourDisplayed(hour)) {
+      return;
+    }
+
+    this.client.waitFor(function() {
+      var where = this._cmpHourDisplayed(hour);
+      if (where === 0) {
+        return true;
+      }
+
+      var moveDelta = 70;
+
+      // element is hidden above the visible part
+      // we need to move it up
+      if (where < 0) {
+        this.actions.flick(
+          this.main,
+          100, 0, 100, moveDelta
+        ).perform();
+      }
+      // element is hidden below the visible part.
+      // we need to move it down
+      if (where > 0) {
+        this.actions.flick(
+          this.main,
+          100, moveDelta, 100, 0
+        ).perform();
+      }
+
+      this.waitForNoScroll();
```

Comment:
If we removed the check before the wait, that line could go at the beginning of the wait.